extensions/blocks/contact-form/attributes.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import colorValidator from '../../shared/colorValidator';
+
+export const defaultAttributes = {
+	subject: {
+		type: 'string',
+		default: __( 'A new message from your website', 'jetpack' ),
+	},
+	to: {
+		type: 'string',
+	},
+	submitButtonText: {
+		type: 'string',
+		default: __( 'Submit', 'jetpack' ),
+	},
+	backgroundButtonColor: {
+		type: 'string',
+	},
+	textButtonColor: {
+		type: 'string',
+	},
+	customBackgroundButtonColor: {
+		type: 'string',
+		validator: colorValidator,
+	},
+	customTextButtonColor: {
+		type: 'string',
+		validator: colorValidator,
+	},
+	submitButtonClasses: {
+		type: 'string',
+	},
+	hasFormSettingsSet: {
+		type: 'string',
+		default: null,
+	},
+	customThankyou: {
+		type: 'string',
+		default: '',
+	},
+	customThankyouMessage: {
+		type: 'string',
+		default: '',
+	},
+	customThankyouRedirect: {
+		type: 'string',
+		default: '',
+	},
+};
```

Comment:
Using the Button block means that we can remove the button-related attributes from the parent ones.
This would require a more involved deprecation/migration path (see the other comment), but it's not complicated per se.

You can refer to what I did with Revue, for example: https://github.com/Automattic/jetpack/blob/46f37cae12cd00eb72ceb543fc5bc2c0a3346e3c/extensions/blocks/revue/deprecated/v1/index.js